### ISSUE TITLE
Improve consistency of reports

### ISSUE THREAD

**Issue by Pennycook** [intel/code-base-investigator#147]
### Feature/behavior summary

The default behavior of CBI is to generate all reports and dump them to `stdout`.  Two new reports were added in #84 and #139, and #134 adopted a new style for tables.  The result of all these changes is that things are now pretty busy, often crammed together, and use different styles:

![image](https://github.com/user-attachments/assets/fc876a59-46dd-4f82-9982-d578fd685d45)

### Request attributes

- [ ] Would this be a refactor of existing code?
- [ ] Does this proposal require new package dependencies?
- [ ] Would this change break backwards compatibility?

### Solution description

- [x] Ensure there is at least one blank line between the output of different reports.
- [x] Adopt a consistent style for report headers.
- [ ] Ensure a consistent ordering for reports.

### Additional notes

I'm not sure what the default order should be.  Putting a summary somewhere near the top seems useful.  The length of the file tree and duplicates report are not known _a priori_, so it may make sense to always put those at the end.

We could allow the user to control the order through the order in which the `-R` flags are provided.

**Comment by laserkelvin** [comment]
So I actually think that, perhaps at a later date[^1], it might be better to have summary and the file tree as two separate commands. As you say it's actually quite busy, and at least personally, I like the idea of the file tree being akin to `ls` and using it as such.

When we have both summary and the file tree, it does feel quite overwhelming, and while there are some cases where you might need to see both I don't think most terminal windows will have the real estate.

If we wanted to "perfect" the file tree display, I think it might even require something `curses` adjacent: you would want to have the header sticky at the top, and the legend sticky on the right. 

TL;DR

1. If we are strapped for time, keep it as is because it's not unusable and we can refine it later based off user feedback, possibly with separation of summary and file tree
2. If we have some time or want to avoid another minor release (I don't think this would be breaking), separate summary and file tree as different commands
3. If we more time, maybe look at something like `pytermgui`, and see how much effort it would be to wrap the existing functionality as windows. Knowing some frontend design this is never actually that straightforward so it might be unreasonable, but now would be the time to introduce new dependencies if we need to.

[^1]: IMO it doesn't need to be 2.0.0, but if you think that this would be a big enough change to *need* another formal release, in that case we should roll it into 2.0.0

**Comment by Pennycook** [comment]
> So I actually think that, perhaps at a later date, it might be better to have summary and the file tree as two separate commands. As you say it's actually quite busy, and at least personally, I like the idea of the file tree being akin to ls and using it as such.

I agree that lumping everything into `codebasin` is less-than-ideal.  It wouldn't be hard to expose it as a `codebasin.tree` and `cbi-tree` similar to `codebasin.coverage` and `cbi-cov`.

But when you say you'd like to use it akin to `ls` -- would you expect it to be _fast_?  One of the reasons that the current functionality is exposed as a "report" is that every time you want to generate a tree, it re-runs the complete analysis pipeline and spits out the tree  at the end.

To me, a separate `cbi-tree` command would make more sense if you could run it on the result of some prior analysis, or if CBI in general had some way of caching partial results.  (i have, at one point or another, written prototypes of such things... But actually making such functionality production-ready might require more stable internals.)

> When we have both summary and the file tree, it does feel quite overwhelming, and while there are some cases where you might need to see both I don't think most terminal windows will have the real estate.

We might be able to meet half-way, until we can address the issues I raised above.

1.2.0 defined the default behavior as "all" which actually mapped to ["summary", "clustering"] (because we only had those two reports).   We could deprecate "all" and say that the default behavior (and the interpretation of "all") is ["summary", "clustering"], so if somebody wants to generate a file tree they have to opt-in by running `codebasin -R files`.

**Comment by Pennycook** [comment]
> I don't know if that changes your mind on anything RE: meeting half way?

You've convinced me, and I've opened https://github.com/intel/code-base-investigator/issues/152 to track splitting things out as you suggest.

If it's okay with you, though, I'd like to merge #150 as-is (since it makes changes to _all_ the reports) and then split out `cbi-tree` as a separate script in a follow-up PR.  I think that will be easier than going through and trying to separate things out mid-PR, especially given that I've got a few other things open.
